app/models.py

```diff
@@ -0,0 +1,230 @@
+    user_type = CharField()
+
+    def is_student(self):
+        return User.user_type == "is_student"
+
+    def is_teacher(self):
+        return User.user_type == "teacher"
+
+    def get_user_courses(self):
+        return (Course
+                .select()
+                .join(UserCourseRelationship, on=UserCourseRelationship.course)
+                .where(UserCourseRelationship.user == self)
+                .order_by(Course.id)
+                )
+
+
+class Course(BaseModel):
+    course_id = IntegerField(unique=True)
+    name = CharField()
+    open_date = DateTimeField()
+
+    def get_course_administrators(self):
+        return (User
+                .select()
+                .join(UserCourseRelationship, on=UserCourseRelationship.user)
+                .where((UserCourseRelationship.course == self) & UserCourseRelationship.is_admin))
+
+
+class UserCourseRelationship(BaseModel):
+    user = ForeignKeyField(User, backref='user')
+    course = ForeignKeyField(Course, backref='course')
+    is_admin = BooleanField(default=False)
+
+
+class Lecture(BaseModel):
+    lecture_id = IntegerField(unique=True)
+    course = ForeignKeyField(Course, backref='lectures')
+    date = DateTimeField()
+    subject = CharField()
+
+    def get_participants(self):
+        return (User
+                .select()
+                .join(UserLectureRelationship, on=UserLectureRelationship.user)
+                .where(UserLectureRelationship.lecture == self))
+
+
+class UserLectureRelationship(BaseModel):
+    user = ForeignKeyField(User, backref='user')
+    lecture = ForeignKeyField(Lecture, backref='lecture')
+
+
+# Utility to create the tables
+def create_tables():
+    with database:
+        database.create_tables([User, Course, Lecture, UserCourseRelationship, UserLectureRelationship])
+
+
+# Examples
+def get_object(model, *expressions):
+    try:
+        return model.get(*expressions)
+    except:
+        raise IntegrityError()
+
+
+def add_course(course_id, name, open_date):
+    course = Course(course_id=course_id, name=name, open_date=open_date)
+    if course.save() != 1:
+        raise IntegrityError()
+    return course
+
+
+def add_user(username, fullname, mail_address, password, user_type):
+    user = User(username=username, fullname=fullname, mail_address=mail_address, password=password, user_type=user_type)
+    # Force insert is used because of the Integer id overlapping the normal saved id.
+    if user.save(force_insert=True) != 1:
+        raise IntegrityError()
+    return user
+
+
+def add_lecture(lecture_id, course_id, date, subject):
+    course = get_object(Course, Course.course_id == course_id)
+    print("Got course")
+    print(course)
+    lecture = Lecture(lecture_id=lecture_id, course=course, date=date, subject=subject)
+    if lecture.save(force_insert=True) != 1:
+        raise IntegrityError()
+    return lecture
+
+
+# 1. Add a user to course
+def user_add_to_course(username, course_id, is_admin=False):
```

Comment:
Is it possible to enter these utils into the relevant Model?
For example, this util into `User`/`Course`/`UserCourseRelationship`?